### PR TITLE
sync walk image

### PR DESCRIPTION
**Pull request by Aychihuahuausa**

### CODE REVIEW

File: src/app/main-client/main-client.component.ts [hunk 1]
```diff
@@ -72,10 +77,38 @@ export class MainClientComponent implements OnInit {
...
+      this.svgCreatorService.generatedSVG(x.imageURL || undefined).then(icon => {
+        this.mapService.addMarker({ lat: x.lat, lng: x.lng }, icon, "", false, null, () => {
+          this.openImageModal(x);
+        });
+        this.changeMapCenter({ lat: x.lat, lng: x.lng });
+      });
     })
   }
 
+  private watchClientPosition() {
```
Comment (heshambadawy) [inline src/app/main-client/main-client.component.ts:93]: What is the reason behind whatch client position ?
if we have a new position why not clear the old markers and have only the new one

File: src/app/main-client/walk-image/walk-image.component.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<ion-header>
```
Comment (heshambadawy) [inline src/app/main-client/walk-image/walk-image.component.html:1]: I have a suggestion here to keep me/us be aware of the parts .
 For instance here  u design that new screen , run and test it why on the ticket place the screen shot for that Will be great

File: src/app/services/socket.events.service.ts [hunk 3]
```diff
@@ -3,8 +3,8 @@ import { io } from 'socket.io-client';
 import { environment } from "src/environments/environment";
 import { UserService } from "./user.service";
 import { Subject } from 'rxjs';
-import { NavigationEnd, Router } from "@angular/router";
-import { filter } from "rxjs/operators";
```
Comment (heshambadawy) [inline src/app/services/socket.events.service.ts:7]: Why u need "Router"  here

File: src/app/share/map/map.service.ts [hunk 4]
```diff
@@ -308,7 +314,8 @@ export class MapService {
   getGoogleLatLng(position: Position) {
     return new google.maps.LatLng(position.lat, position.lng);
   }
+  checkIfMarkerDisplayed(marker): boolean {
+    return this.map.getBounds().contains(marker.getPosition());
```
Comment (heshambadawy) [inline src/app/share/map/map.service.ts:318]: if that is the right way to see if the map contains a marker .

and all markers to list and check if that new one exist

File: src/app/main-client/main-client.component.ts [hunk 5]
```diff
@@ -72,10 +77,38 @@ export class MainClientComponent implements OnInit {
       this.footerId = 1;
       this.mainClientService.clearCurrentRequest();
       this.mapService.clearMap();
-      this.setCenter();
+      this.getCurrentPostion();
+    });
+
+    this.socketEventsService.newImageTakenEvent.subscribe((x: WalkImage) => {
+      this.svgCreatorService.generatedSVG(x.imageURL || undefined).then(icon => {
```
[the remark this answers is not in the document]
Comment (Aychihuahuausa) [reply src/app/main-client/main-client.component.ts:84]: I tested it and this was Alex quest

File: src/app/main-client/main-client.component.ts (hunk 1 above)
Comment (Aychihuahuausa) [reply src/app/main-client/main-client.component.ts:93]: Alex want to update client position too and I update marker's position not adding a new one

File: src/app/services/socket.events.service.ts (hunk 3 above)
Comment (Aychihuahuausa) [reply src/app/services/socket.events.service.ts:7]: done

File: src/app/share/map/map.service.ts (hunk 4 above)
Comment (Aychihuahuausa) [reply src/app/share/map/map.service.ts:318]: what is the problem here

**Pull request merged**